lib/email_reply_parser.rb

```diff
@@ -134,6 +138,33 @@ def read(text)
     EMPTY = "".freeze
     SIGNATURE = '(?m)(--\s*$|__\s*$|\w-$)|(^(\w+\s+){1,3}ym morf tneS$)'
     SIG_REGEX = Regexp.new(SIGNATURE)
+    MULTILINE_QUOTE_HEADER_EXPRESSIONS = [
+      # English
+      /(^On\s((?!On).)*wrote:$)$/m,
+      # Spanish
+      /(?!El.*El\s.+?escribi.*\s*:)(El\s(.+?)escribi.*\s*:)/m,
+      # French
+      /(?!Le.*Le\s.+?a .*crit[a-zA-Z0-9.:;<>()&@ -]*:)(Le\s(.+?)a .*crit[a-zA-Z0-9.:;<>()&@ -]*:)/m,
+      # German
+      /(?!Am.*Am\s.+?schrieb.*:)(Am\s(.+?)schrieb.*:)/m,
+      # Italian
+      /(?!Il.*Il\s.+?scritto.*:)(Il[\s\S]+?(.+?)scritto:)/m
+    ]
+    QUOTE_HEADER_REGULAR_EXPRESSIONS = [
+      # English
+      /^On.*wrote:$/,
+      /^(From|Sent|To|Subject):.*$/m,
+      # Spanish
+      /\s*El.*a las.*escribi.*\s*:$/,
+      /^(De|Enviado|Para|Asunto):.*$/m,
+      # French
+      /Le.*a .*crit.*[> ]:$/m,
+      /^(De|A|Envoyé|Objet)\s*:.*$/m,
+      # German
+      /[a-zA-Z]{2,5}.*schrieb.*:$/,
+      # Italian
+      /^Il.+?ha scritto:$/
+    ]
```

Comment:
:speech_balloon: You can freeze all these constants

Reply:
Done!